File: t/VB/OmrProcessingDemo/UI/OMRProcessInputWizard/TemplatePanel.vb
Imports Leadtools.Codecs
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports Leadtools
Imports Leadtools.Forms.Processing.Omr
Imports System.IO
Imports Leadtools.Controls

Partial Public Class TemplatePanel
   Inherits WizardStep

   Private _iv As ImageViewer
   Private _templateImage As RasterImage
   Private _fileImage As RasterImage
   Private _templateEditorEngine As ITemplateForm
   Private _loadedEditorEngine As ITemplateForm

   Public ReadOnly Property LoadedEditor As ITemplateForm
      Get
         Return If(rdbtnTemplate.Checked, _templateEditorEngine, _loadedEditorEngine)
      End Get
   End Property

   Public ReadOnly Property LoadedImage As RasterImage
      Get
         Return _iv.Image
      End Get
   End Property

   Public Sub New()
      InitializeComponent()
      _iv = New ImageViewer()
      _iv.Dock = DockStyle.Fill
      _iv.AutoDisposeImages = False
      AddHandler _iv.ItemChanged, AddressOf Riv_ItemChanged
      pnlThumbnail.Controls.Add(_iv)
   End Sub

   Public Sub New(ByVal defaultImage As RasterImage, ByVal currentEngine As ITemplateForm)
      _templateImage = defaultImage
      _templateEditorEngine = currentEngine

      If defaultImage IsNot Nothing Then
         rdbtnTemplate.Checked = True
      Else
         rdBtnFile.Checked = True
         rdbtnTemplate.Enabled = False
      End If
   End Sub

   Private Sub rdBtnToggled(ByVal sender As Object, ByVal e As EventArgs)
      Try

         If rdBtnFile.Checked Then
            _iv.Image = _fileImage
         End If

         If rdbtnTemplate.Checked Then
            _iv.Image = _templateImage
         End If

      Catch __unusedException1__ As Exception
         _iv.Image = Nothing
      End Try

      txtFilePath.Enabled = rdBtnFile.Checked
      btnBrowse.Enabled = rdBtnFile.Checked
      OnCanProceed()
   End Sub

   Private Sub btnTemplateBrowse_Click(ByVal sender As Object, ByVal e As EventArgs)
      Dim input As String = ""

      If MainForm.ChooseTemplate(input) Then
         LoadTemplate(input)
         _fileImage = _loadedEditorEngine.Pages(0).Image
         _iv.Image = _fileImage
         txtFilePath.Text = input
         OnCanProceed()
      End If
   End Sub

   Protected Overrides Sub OnCanProceed()
      OnCanProceed(LoadedEditor IsNot Nothing AndAlso _iv.Image IsNot Nothing)
   End Sub

   Private Sub LoadTemplate(ByVal selectedPath As String)
      _loadedEditorEngine = MainForm.GetOmrEngine().CreateTemplateForm()
      _loadedEditorEngine.Load(New FileStream(selectedPath, FileMode.Open, FileAccess.Read))
   End Sub

   Private Sub Riv_ItemChanged(ByVal sender As Object, ByVal e As ImageViewerItemChangedEventArgs)
      If e.Reason = ImageViewerItemChangedReason.Image Then
         _iv.Zoom(ControlSizeMode.Fit, 1, LeadPoint.Empty)
      End If
   End Sub
End Class

File: t/CS/OmrProcessingDemo/UI/OMRProcessInputWizard/TemplatePanel.cs
using Leadtools.Codecs;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Leadtools;
using Leadtools.Forms.Processing.Omr;
using System.IO;
using Leadtools.Controls;

namespace OmrProcessingDemo.UI
{
   public partial class TemplatePanel : WizardStep
   {
      private ImageViewer iv;

      private RasterImage templateImage;
      private RasterImage fileImage;

      private ITemplateForm templateEditorEngine;
      private ITemplateForm loadedEditorEngine;

      public ITemplateForm LoadedEditor { get { return rdbtnTemplate.Checked ? templateEditorEngine : loadedEditorEngine; } }

      public RasterImage LoadedImage { get { return iv.Image; } }

      public TemplatePanel()
      {
         InitializeComponent();

         iv = new ImageViewer();
         iv.Dock = DockStyle.Fill;
         iv.AutoDisposeImages = false;
         iv.ItemChanged += Riv_ItemChanged;
         pnlThumbnail.Controls.Add(iv);
      }

      public TemplatePanel(RasterImage defaultImage, ITemplateForm currentEngine) : this()
      {
         templateImage = defaultImage;
         templateEditorEngine = currentEngine;

         if (defaultImage != null)
         {
            rdbtnTemplate.Checked = true;
         }
         else
         {
            rdBtnFile.Checked = true;
            rdbtnTemplate.Enabled = false;
         }
      }

      private void rdBtnToggled(object sender, EventArgs e)
      {
         try
         {
            if (rdBtnFile.Checked)
            {
               iv.Image = fileImage;
            }
            if (rdbtnTemplate.Checked)
            {
               iv.Image = templateImage;
            }
         }
         catch (Exception)
         {
            iv.Image = null;
         }

         txtFilePath.Enabled = rdBtnFile.Checked;
         btnBrowse.Enabled = rdBtnFile.Checked;

         OnCanProceed();
      }

      private void btnTemplateBrowse_Click(object sender, EventArgs e)
      {
         string input = "";
         if (MainForm.ChooseTemplate(out input))
         {
            LoadTemplate(input);
            fileImage = loadedEditorEngine.Pages[0].Image;
            iv.Image = fileImage;
            txtFilePath.Text = input;

            OnCanProceed();
         }
      }

      protected override void OnCanProceed()
      {
         OnCanProceed(LoadedEditor != null && iv.Image != null);
      }

      private void LoadTemplate(string selectedPath)
      {
         loadedEditorEngine = MainForm.GetOmrEngine().CreateTemplateForm();

         loadedEditorEngine.Load(new FileStream(selectedPath, FileMode.Open, FileAccess.Read));
      }
      private void Riv_ItemChanged(object sender, ImageViewerItemChangedEventArgs e)
      {
         if (e.Reason == ImageViewerItemChangedReason.Image)
         {
            iv.Zoom(ControlSizeMode.Fit, 1, LeadPoint.Empty);
         }
      }

   }
}